meson.build

```diff
@@ -181,7 +182,7 @@ cc_warning_flags = cc.get_supported_arguments(cc_warning_flags_test)
 cc_flags = [cc_debug_flags, cc_os_flags, cc_warning_flags]
 add_project_arguments(cc_flags, language : 'c')
 
-incdir = include_directories('src/includes')
+incdir = include_directories('src/common')
```

Comment:
i don't think collapsing includes/ into common/ is a good pattern.  having a dir dedicated to headers for APIs that are meant to be reused across the codebase helps enforce good boundaries.  when the dir mixes such "exported" APIs with internal headers, it's hard for other dirs to know what's "internal", and what should be kept that way.

basically, the existence of a header file in a dir should not be a signal by itself that it's OK for reuse outside of the dir.